.drone.yml

```diff
@@ -190,27 +190,45 @@ steps:
       - git lfs install --skip-repo
       # We should be in a cached git repo, but clone if the cache failed
       - "[ -d .git ] || (echo 'get-cached-staging failed: checkout will be SLOW and LFS $$$!' && git clone --branch $DRONE_BRANCH $DRONE_GIT_HTTP_URL .)"
-      # Checkout our target branch (the PR's branch).
+      # Checkout our source branch (the PR's branch).
       - git fetch origin $DRONE_SOURCE_BRANCH
       - git checkout $DRONE_SOURCE_BRANCH
       - git pull
       - git reset --hard $DRONE_COMMIT
-      # Also fetch the target branch (which is staging or staging-next for pull requests.) We need this for determining which tests to run based on changed files.
+      # Also fetch the target branch (usually staging for pull requests). We
+      # need this for determining which tests to run based on changed files.
       - git remote set-branches --add origin $DRONE_TARGET_BRANCH
       - git fetch --depth 100 origin $DRONE_TARGET_BRANCH
       # Merge so we're up-to-date with the target before running tests.
       - git config user.name "Drone"
       - git config user.email "drone-fake-user@code.org"
       - git merge origin/$DRONE_TARGET_BRANCH
 
-  - name: restore-cache
+  # We cache staging's rbenv and database in the
+  # `cache-staging-build-artifacts` pipeline.
+  - name: get-cached-staging-rbenv
     image: plugins/s3-cache
+    volumes:
+      - name: rbenv
+        path: /drone/src/home/ci/.rbenv
     settings:
+      filename: cache-staging-rbenv.tar.gz
       endpoint: s3://cdo-drone
+      restore: true
       debug: true
       pull: true
+
+  - name: get-cached-staging-mysql
+    image: plugins/s3-cache
+    volumes:
+      - name: mysql
+        path: /drone/src/var/lib/mysql
+    settings:
+      filename: cache-staging-mysql.tar.gz
+      endpoint: s3://cdo-drone
       restore: true
-      filename: ui-tests-cache-ubuntu-20.tar.gz
+      debug: true
+      pull: true
```

Comment:
See https://github.com/code-dot-org/code-dot-org/pull/66065 for an alternative way we could exchange the cached data between these steps and the `ui-tests` step without relying on extra volumes. I'd love to get y'all's thoughts on which way you'd prefer to go!